tests/test_matrix_utils.py

```
@@ -3711,11 +3937,88 @@ async def test_connect_matrix_credentials_load_exception_uses_config(monkeypatch
+    """Explicit credentials_path should be expanded and used first."""
+    mock_client = MagicMock()
+    mock_client.rooms = {}
+    mock_client.sync = AsyncMock(return_value=SimpleNamespace())
+    mock_client.should_upload_keys = False
+    mock_client.get_displayname = AsyncMock(
+        return_value=SimpleNamespace(displayname="Bot")
+    )
+    mock_client.close = AsyncMock()
+    mock_client.restore_login = MagicMock()
+
+    expanded_path = "/tmp/explicit_credentials.json"
```

Ruff (0.14.14)

Probable insecure usage of temporary file or directory: "/tmp/explicit_credentials.json"

(S108)